buildSrc/src/main/java/quilt/internal/tasks/lint/SpellingChecker.java

```diff
@@ -0,0 +1,109 @@
+import cuchaz.enigma.translation.representation.AccessFlags;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SpellingChecker implements Checker<Entry<?>> {
+    private static final String PACKAGE_INFO_CLASS_PACKAGE = "net/minecraft/unused/packageinfo/";
+
+    private static final Set<String> ALLOWED_WORDS = new HashSet<>();
+    static {
+        // collect allowed words
+        URL url;
+        try {
+            url = new URL("https://raw.githubusercontent.com/ix0rai/qm-base-allowed-wordlist/main/allowed_english_words.txt");
+            Set<String> englishWords = new java.util.Scanner(url.openStream()).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
+            ALLOWED_WORDS.addAll(englishWords);
+        } catch (Exception e) {
+            throw new RuntimeException("failed to download and read english word file for spell check in mapping lint task!");
+        }
+
+        // collect minecraft words
+        InputStream parsedFile = SpellingChecker.class.getClassLoader().getResourceAsStream("minecraft_specific_words.txt");
+        if (parsedFile == null) {
+            throw new RuntimeException("could not find minecraft word file for spelling check in mapping lint task!");
+        }
+
+        Set<String> minecraftWords = getLines(parsedFile);
+        // ignore comments and empty lines
+        Set<String> minecraftWordsCopy = Set.copyOf(minecraftWords);
+        for (String word : minecraftWordsCopy) {
+            if (word.isEmpty() || word.startsWith("//")) {
+                minecraftWords.remove(word);
+            }
+        }
+
+        ALLOWED_WORDS.addAll(minecraftWords);
+    }
+
+    private static Set<String> getLines(InputStream stream) {
+        return new java.util.Scanner(stream).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
+    }
+
+    @Override
+    public void check(Entry<?> entry, EntryMapping mapping, Function<Entry<?>, AccessFlags> accessProvider, ErrorReporter errorReporter) {
+        String name = mapping.targetName();
+
+        if (name != null
+                && !name.startsWith(PACKAGE_INFO_CLASS_PACKAGE)
+                // ignore unmapped methods and classes
+                && !(entry instanceof MethodEntry && name.startsWith("m_"))
+                && !(entry instanceof ClassEntry && name.startsWith("C_"))) {
+            Set<String> words = new HashSet<>();
+
+            // fixme optimise!
+            // split by slashes
+            String[] slashSeparatedWords = name.split("/");
```

Comment:
Since only classes can have slashes in their target names, you could do something like this
```java
String[] packageNames = new String[0];
if (entry instanceof ClassEntry) {
  name = name.substring(name.lastIndexOf('/') + 1);
  packageNames = name.substring(0, name.lastIndexOf('/'));
}

words.addAll(List.of(packageNames));
// ...

String[] underscoreSeparatedWords = name.split("_");
```

Reply:
packages can have underscores in the names, so the underscore filter is still required on them

Reply:
Ah, right. What if you put them into a collection alongside the name, and then handle underscores on that collection?

Reply:
good idea; I'll implement that!
any optimisation is important